src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,39 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const booksSlice = createSlice({
+  name: 'books',
+  initialState: {
+    items: [
+      {
+        id: 'item1',
+        title: 'The Great Gatsby',
+        author: 'John Smith',
+        category: 'Fiction',
+      },
+      {
+        id: 'item2',
+        title: 'Anna Karenina',
+        author: 'Leo Tolstoy',
+        category: 'Fiction',
+      },
+      {
+        id: 'item3',
+        title: 'The Selfish Gene',
+        author: 'Richard Dawkins',
+        category: 'Nonfiction',
+      },
+    ],
+  },
+  /* eslint no-param-reassign: ["error", { "props": false }] */
+  reducers: {
+    addBook: (state, action) => {
+      state.items.push(action.payload);
+    },
```

Comment:
Well done overall but the `addBook` reducer is also modifying the `state `object directly by using the `push `method. 🗯️  To avoid this, you can create a new array with the updated items and return a new object with the updated items array. 📦  Here's how you can modify the `addBook` reducer:
```
addBook: (state, action) => {
  const newItem = action.payload;
  const updatedItems = [...state.items, newItem];
  return { ...state, items: updatedItems };
},
```
In the updated reducer, we create a new `newItem` variable 🆕  that holds the `payload` value from the `action`. We then create a new array `updatedItems` that spreads the existing `state.items` array and adds the new item to the end of the array using the `spread` operator. Finally, 🤔  we return a new object with the original state properties spread using the spread operator `...state`, and overwrite the `items` property with the `updatedItems` array.

This will achieve your goal of avoiding direct modification of the state object in both reducers. 🤓

Reply:
Thanks, man. Much appreciated.